spec/gorouter_templates_spec.rb

```diff
@@ -777,6 +781,231 @@ def make_properties(link, link_property)
+
+            getters = property_chain[0..-2]
+            setter = property_chain.last
+
+            set_this = hash
+            getters.each do |getter|
+              getter = getter.to_i if set_this.class == Array
+              set_this = set_this.fetch(getter)
+            end
+
+            set_this.delete(setter)
+          end
+
+          context 'when the link is not provided' do
+            context 'when the property is set' do
+              before do
+                # TODO: constant it
+                set_at_property(deployment_manifest_fragment, link_config.property, property_value)
+              end
+
+              it 'should prefer the value in the properties' do
+                expect(get_at_property(parsed_yaml, link_config.parsed_yaml_property)).to eq(property_value)
+              end
+            end
+
+            context 'when the property is not set' do
+              before do
+                delete_at_property(deployment_manifest_fragment, link_config.property)
+              end
+
+              it 'should error' do
+                expect do
+                  parsed_yaml
+                end.to raise_error(
+                  RuntimeError,
+                  "#{link_config.description} not found in properties nor in \"#{link_config.link_namespace}\" link. This value can be specified using the \"#{link_config.property}\" property."
+                )
+              end
+            end
+          end
+
+          context 'when the link is provided' do
+            def make_properties(link, link_property)
+              property_chain = link.split('.').reverse
+
+              property_chain.reduce(link_property) do |memo, obj|
+                { obj => memo }
+              end
+            end
+
+            let(:links) do
+              [
+                Bosh::Template::Test::Link.new(
+                  name: link_config.link_namespace,
+                  properties: make_properties(link_config.link, link_value)
+                )
+              ]
+            end
+
+            let(:rendered_template) { template.render(deployment_manifest_fragment, consumes: links) }
+
+            context 'when the property is set' do
+              before do
+                set_at_property(deployment_manifest_fragment, link_config.property, property_value)
+              end
+
+              it 'should prefer the value in the properties' do
+                expect(get_at_property(parsed_yaml, link_config.parsed_yaml_property)).to eq(property_value)
+              end
+            end
+
+            context 'when the property is not set' do
+              before do
+                delete_at_property(deployment_manifest_fragment, link_config.property)
+              end
+
+              it 'should render the value from the link' do
+                expect(get_at_property(parsed_yaml, link_config.parsed_yaml_property)).to eq(link_value)
+              end
+            end
+          end
+        end
```

Comment:
I anticipate a reviewer will wonder why I duplicated this from further down in the file. I've tried moving this code several times and kept hitting problems with the scoping of `deployment_manifest_fragment` and how shared examples work.

Reply:
👍